Bottom: 2d3dccdafae81e0832f780991054975f627e0fc4
Top:    8a9c77f483164bbbfd2673cb9bac4b5dca080839
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-02-16 22:40:09 +1300

Refresh of define-stream-result

---

diff --git a/testtools/testresult/real.py b/testtools/testresult/real.py
index 34d97f0..98cc7d3 100644
--- a/testtools/testresult/real.py
+++ b/testtools/testresult/real.py
@@ -341,26 +341,30 @@ class StreamResult(object):
         """Inform the result about a test status.
 
         :param test_id: The test whose status is being reported.
-        :param test_status: One of:
-        * exists - the test exists. This is used when a test is not being
-          executed. Typically this is when querying what tests could be run in
-          a test run (which is useful for selecting tests to run).
-        * inprogress - the test is currently running. Emitted by tests when
-          they start running and at any intermediary point they might choose to
-          indicate their continual operation.
-        * xfail - the test failed but that was expected. This is purely
-        * informative - the test is not considered to be a failure.
-        * uxsuccess - the test passed but was expected to fail. The test will
-          be considered a failure.
-        * finish - the test has finished. If no error status had been reported,
-          it can now be considered a success, otherwise it should be considered
-          a failure. This must be the last event sent for this test (including
-          file events).
-        * fail - the test failed (or errored). The test will be considered a
-          failure.
-        * skip - the test was selected to run but chose to be skipped. E.g.
-          a test dependency was missing. This is purely informative - the
-          test is not considered to be a failure.
+        :param test_status: The status for the test. There are two sorts of
+            status - interim and final status events. As many interim events
+            can be generated as desired, but only one final event. After a
+            final status event any further file or status events from the
+            same test_id+route_code may be discarded or associated with a new
+            test by the StreamResult. (But no exception will be thrown).
+            Interim states:
+            * inprogress - the test is currently running. Emitted by tests when
+              they start running and at any intermediary point they might
+              choose to indicate their continual operation.
+            Final states:
+            * exists - the test exists. This is used when a test is not being
+              executed. Typically this is when querying what tests could be run
+              in a test run (which is useful for selecting tests to run).
+            * xfail - the test failed but that was expected. This is purely
+              informative - the test is not considered to be a failure. 
+            * uxsuccess - the test passed but was expected to fail. The test
+              will be considered a failure.
+            * success - the test has finished without error.
+            * fail - the test failed (or errored). The test will be considered
+              a failure.
+            * skip - the test was selected to run but chose to be skipped. E.g.
+              a test dependency was missing. This is purely informative - the
+              test is not considered to be a failure.
         :param test_tags: Optional set of tags to apply to the test. Tags
             have no intrinsic meaning - that is up to the test author.
         :param runnable: Allows status reports to mark that they are for
